pkg/cfn/builder/nodegroup.go

```diff
@@ -262,12 +265,57 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 		)
 	}
 
+	// not using aws.IntValue because it's not a required field
+	// and aws.IntValue would result in 0 if it's nil.
+	if n.spec.DesiredCapacity != nil && *n.spec.DesiredCapacity == 0 {
+		clusterTags, err := generateClusterAutoscalerTags(n.spec)
+		if err != nil {
+			return err
+		}
+		tags = append(tags, clusterTags...)
```

Comment:
what if `len(tags)` is greater than 50 here, after they have been appended to the rest of the tags? maybe that validation should be moved here.

Reply:
Oh snap! You're right!! :O

Reply:
Frigging love you Niki. Seriously.